src/Service/FilesystemFactory.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace App\Service;
+
+use App\Classes\LocalCachingFilesystemDecorator;
+use Aws\S3\S3Client;
+use League\Flysystem\AdapterInterface;
+use League\Flysystem\AwsS3v3\AwsS3Adapter;
+use League\Flysystem\Adapter\Local;
+use League\Flysystem\Cached\CachedAdapter;
+use League\Flysystem\Cached\Storage\Adapter;
+use League\Flysystem\Cached\Storage\Memory as MemoryStore;
+use League\Flysystem\Filesystem as LeagueFilesystem;
+use League\Flysystem\FilesystemInterface;
+
+class FilesystemFactory
+{
+    const LOCAL_S3_CACHE_DIR = '/ilios/s3-cache';
+
+    /**
+     * @var Config
+     */
+    protected $config;
+
+    /**
+     * @var string
+     */
+    private $kernelCacheDir;
+
+    public function __construct(
+        Config $config,
+        string $kernelCacheDir
+    ) {
+        $this->config = $config;
+        $this->kernelCacheDir = $kernelCacheDir;
+    }
+
+    public function getFilesystem() : FilesystemInterface
```

Comment:
this is more of a comment than anything - if you were to stick more strictly to the Factory design pattern, then the type of FS would have to be passed in to this method as argument.

Reply:
That makes sense, Symfony sets the interface for us though so it has to be this way here.